audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDeviceSelector.kt

```diff
@@ -117,9 +121,9 @@ class AudioDeviceSelector {
      * longer needed, [AudioDeviceSelector.stop] should be called in order to prevent a
      * memory leak.
      *
-     * @param listener receives audio device change events
+     * @param listener optional, receives audio device change events
      */
-    fun start(listener: AudioDeviceChangeListener) {
+    fun start(listener: AudioDeviceChangeListener? = null) {
```

Comment:
Why are we allowing this to be nullable now?

Reply:
I believe this is what @idelgado intended in the design, in that there are 2 ways users can retrieve the available and selected devices.

1) They pass in a listener when calling start to subscribe to changes

2) They don't pass in a listener, and can retrieve the available audio devices and selected devices manually later by calling availableAudioDevices or selectedAudioDevice from the AudioDeviceSelector

Reply:
Point number 2 is true whether this argument is allowed to be null or not. I don't think we should allow null because we have to account for the behavior when the developer provides a null listener. @idelgado 's [original code explicitly prohibits null](https://github.com/idelgado/voice-quickstart-android/blob/task/device-selection/app/src/main/java/com/twilio/voice/quickstart/audio/AudioDeviceSelector.java#L86) which I think is the correct behavior.

Reply:
True, you can access the properties in both scenarios. It's honestly not a big deal, but I thought it was a nicer experience as an end user to not have to type an empty lambda `start {_,_->}` whenever I just wanted to invoke `start()` without subscribing.

Reply:
Yeah it's a tradeoff. Depending on how many spots you use the listener internally it may not be a major issue to allow for `null` but in general I like making thing `nonnull`. Makes for a stronger API contract and less boilerplate internally.